zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/RequestMapper.java

```diff
@@ -373,14 +375,21 @@ public static Either<ProblemDetail, BroadcastSignalRequest> toBroadcastSignalReq
   public static Authentication getAuthentication() {
     final List<String> authorizedTenants = TenantAttributeHolder.tenantIds();
 
-    final String token =
+    final var token =
         Authorization.jwtEncoder()
             .withIssuer(JwtAuthorizationBuilder.DEFAULT_ISSUER)
             .withAudience(JwtAuthorizationBuilder.DEFAULT_AUDIENCE)
             .withSubject(JwtAuthorizationBuilder.DEFAULT_SUBJECT)
-            .withClaim(Authorization.AUTHORIZED_TENANTS, authorizedTenants)
-            .encode();
-    return new Builder().token(token).tenants(authorizedTenants).build();
+            .withClaim(Authorization.AUTHORIZED_TENANTS, authorizedTenants);
+
+    final var authenticatedUser =
+        (CamundaUser) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+
+    if (authenticatedUser != null) {
+      token.withClaim(Authorization.AUTHORIZED_USER_KEY, authenticatedUser.getUserKey());
+    }
+
+    return new Builder().token(token.build()).tenants(authorizedTenants).build();
```

Comment:
❓ Do we have a followup issue to not put this in a JWT anymore?

Reply:
There isn't a dedicated issue at the moment but I will sort that out